### PR TITLE
Stub implementation for file upload handler

### LINKED ISSUE #49: Write FileUpload handler (stub)

### PR DESCRIPTION
**Pull request by jcoyne**
Blocked by #87
Fixes #49

- [x] document how to add a file
- [x] return correct response
- [x] write to s3
- [x] write file record to dynamodb
- [x] unit test (happy path and error paths)
- [x] integration test
- [x] handles errors

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+// S3BucketStorage provides file storage using an S3 bucket
+type S3BucketStorage struct {
+	config   *config.Config
+	uploader *s3manager.Uploader
+}
+
+// NewS3Bucket creates a new S3 bucket storage
+func NewS3Bucket(config *config.Config) Storage {
```
Comment (cmh2166) [inline storage/storage.go]: Do we need to create a new S3 Bucket here? I think we rather have the S3 bucket defined before and TACO would not be created new buckets, just passing files to the existing. Also, I don't think this method is use (afaict) so maybe just remove to make clear TACO shouldn't be creating new buckets.

**Comment by cmh2166** [comment]
Also looks like it needs a rebase.

File: storage/storage.go (hunk 1 above)
Comment (jcoyne) [reply storage/storage.go]: This doesn't create a new bucket in s3, it creates a new instance of S3Bucket which has the method to upload a file.

Comment (cmh2166) [reply storage/storage.go]: gotcha, thanks for the clarification

**Pull request merged**